ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -17862,44 +17865,43 @@ void HqlCppTranslator::doBuildHttpHeaderStringFunction(BuildCtx &ctx, IHqlExpres
 
 }
 
+
 ABoundActivity * HqlCppTranslator::doBuildActivitySOAP(BuildCtx & ctx, IHqlExpression * expr, bool isSink, bool isRoot)
 {
     ThorActivityKind tak;
-    const char * helper;
     unsigned firstArg = 0;
-    IHqlExpression * dataset = NULL;
+    IHqlExpression * dataset = nullptr;
     Owned<ABoundActivity> boundDataset;
     IHqlExpression * selSeq = querySelSeq(expr);
-    if (expr->getOperator() == no_newsoapcall)
+
+    bool isHttpPost = false;
+    const char * helper= (isSink) ? "SoapAction" : "SoapCall";
+
+    switch (expr->getOperator())
     {
-        if (isSink)
-        {
-            tak = TAKsoap_rowaction;
-            helper = "SoapAction";
-        }
-        else
-        {
-            tak = TAKsoap_rowdataset;
-            helper = "SoapCall";
-        }
-    }
-    else
+    case no_new_httppost:
+        isHttpPost = true; //fall through
+    case no_newsoapcall:
+        tak = (isSink) ? TAKsoap_rowaction : TAKsoap_rowdataset;
+        break;
+    case no_httppost:
+    case no_httppost_ds:
+    case no_new_httppost_ds:
+    case no_httpaction_ds:
+    case no_new_httpaction_ds:
+        isHttpPost = true; //fall through
```

Comment:
probably needs fallthrough as all one word, and on a separate line (to not trigger error on gcc 8.3+)